src/NuGet.Indexing/ResponseFormatter.cs

```diff
@@ -346,7 +346,19 @@ public static void WriteStatsResult(JsonWriter jsonWriter, NuGetIndexSearcher se
             jsonWriter.WriteStartObject();
             WriteProperty(jsonWriter, "numDocs", searcher.IndexReader.NumDocs());
```

Comment:
I think this would look a lot cleaner if `SearcherManager` had a dictionary mapping property name to value, i.e.
```
{ numDocs, () => IndexReader.NumDocs()},
{ indexName, () => Manager.IndexName},
...
```

This code would be reduced significantly, and it would be very easy to add additional diagnostics.
```
foreach (var diagnostic in searcher.Diagnostics) {
    WriteProperty(jsonWriter, diagnostic.Name, diagnostic.GetValue());
}
```